### ISSUE TITLE
Look for common implementation of WebDriver-like executors

### ISSUE THREAD

**Issue by jgraham** [w3c/web-platform-tests#10036]
With the refactor of the `Protocol` object, it's starting to look like we could have a common `TestHarnessExecutor` differing only in the `Protocol` object.

**Comment by foolip** [comment]
I have had the same thought. Would this apply to RefTestExecutor as well?

**Comment by foolip** [comment]
@kereliuk, WDYT?

**Comment by jgraham** [comment]
Yes, it would apply to reftest executors, but they aren't fully covered by the `Protocol` abstraction yet.
